src/modules/hip/kernel/look_up_table.cpp

```diff
@@ -1,127 +1,224 @@
+                                               unsigned int *yroi_end,
+                                               unsigned int *height,
+                                               unsigned int *width,
+                                               unsigned int *max_width,
+                                               unsigned long long *batch_index,
+                                               const unsigned int channel,
+                                               unsigned int *inc, // use width * height for pln and 1 for pkd
+                                               const int plnpkdindex) // use 1 pln 3 for pkd
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if (id_x < width[id_z] && id_y < height[id_z])
+    {
+        long pixIdx = batch_index[id_z] + (id_x + id_y * max_width[id_z]) * plnpkdindex;
+
+        if ((id_y >= yroi_begin[id_z]) && (id_y <= yroi_end[id_z]) && (id_x >= xroi_begin[id_z]) && (id_x <= xroi_end[id_z]))
+        {
+            for (int indextmp = 0; indextmp < channel; indextmp++)
+            {
+                int luptrIndex = (id_z * channel * 256) + (input[pixIdx] * plnpkdindex);
+                output[pixIdx] = saturate_8u((int) lutPtr[luptrIndex]);
+                pixIdx += inc[id_z];
+            }
+        }
+        else if ((id_x < width[id_z]) && (id_y < height[id_z]))
+        {
+            for (int indextmp = 0; indextmp < channel; indextmp++)
+            {
+                output[pixIdx] = input[pixIdx];
+                pixIdx += inc[id_z];
+            }
+        }
     }
-  }
 }
 
-extern "C" __global__ void look_up_table_batch_tensor(
-    unsigned char *input, unsigned char *output,
-    unsigned char *lutPtr, unsigned int *height,
-    unsigned int *width, unsigned int *max_width,
-    unsigned long *batch_index, const unsigned int channel,
-    unsigned int *inc,
-    unsigned int *dst_inc, // use width * height for pln and 1 for pkd
-    const int in_pln_pkd_ind, const int out_pln_pkd_ind // use 1 pln 3 for pkd
-) {
-  int id_x = hipBlockIdx_x *hipBlockDim_x + hipThreadIdx_x;
-  int id_y = hipBlockIdx_y *hipBlockDim_y + hipThreadIdx_y;
-  int id_z = hipBlockIdx_z *hipBlockDim_z + hipThreadIdx_z;
-  if (id_x < width[id_z] && id_y < height[id_z]) {
-    long in_pix_index =
-        batch_index[id_z] + (id_x + id_y * max_width[id_z]) * in_pln_pkd_ind;
-    long out_pix_index =
-        batch_index[id_z] + (id_x + id_y * max_width[id_z]) * out_pln_pkd_ind;
-    int luptrIndex = id_z << 8;
-    for (int indextmp = 0; indextmp < channel; indextmp++) {
-      int lutIndex = luptrIndex + input[in_pix_index];
-      output[out_pix_index] = lutPtr[lutIndex];
-      in_pix_index += inc[id_z];
-      out_pix_index += dst_inc[id_z];
+extern "C" __global__ void look_up_table_batch_tensor(unsigned char *input,
```

Comment:
I don't think the 'tensor' word appears in the lower level files. Lets rename:
- Kernel name @L97: look_up_table_batch_tensor -> lut_batch
- Kernel name @L129: look_up_table_batch_tensor_int8 -> lut_batch_int8
- Kernel call @L173: look_up_table_batch_tensor -> lut_batch
- Kernel call @L204: look_up_table_batch_tensor_int8 -> lut_batch_int8
Try for build issues.

Reply:
Abishek, I tried to do it, but it  is throwing an """ conflicting types for 'look_up_table_batch' """